### PR TITLE
Initial docker

### PR DESCRIPTION
**Pull request by brylie**
Work-in-progress

### CODE REVIEW

**Comment by brylie** [comment]
Currently getting an error when running `docker-compose up`

> python: error while loading shared libraries: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
ERROR: Service 'api' failed to build: The command '/bin/sh -c python -m pip install pipenv' returned a non-zero code: 127

File: api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM python:3.7
+
+LABEL Author="Brylie Christopher Oxley"
+LABEL E-mail="brylie.oxley@maas.global"
+LABEL version="0.0.1"
+
+LABEL Name=sustainable-mobility-api Version=0.0.1
+EXPOSE 8080
+
+WORKDIR /app
```
Comment (MarkusShepherd) [inline api/Dockerfile]: I'd say it's better to make sure the dir `/app` actually exists:

```Dockerfile
RUN mkdir --parents /app
```

File: api/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ADD . /app
```
Comment (MarkusShepherd) [inline api/Dockerfile]: Use [`COPY` instead of `ADD`](https://www.ctl.io/developers/blog/post/dockerfile-add-vs-copy/)

File: api/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+FROM python:3.7
+
+LABEL Author="Brylie Christopher Oxley"
+LABEL E-mail="brylie.oxley@maas.global"
+LABEL version="0.0.1"
```
Comment (huksley) [inline api/Dockerfile]: Could it be more clear to users if we synchronize version with estimator 0.1.2?

File: api/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+LABEL Name=sustainable-mobility-api Version=0.0.1
+EXPOSE 8080
+
+WORKDIR /app
+ADD . /app
+
+RUN python -m pip install pipenv
+RUN pipenv install --ignore-pipfile
```
Comment (MarkusShepherd) [inline api/Dockerfile]: I'm not sure `--ignore-pipfile` is really necessary here, I'd use`--deploy` instead (since that's what we're doing)

**Comment by brylie** [comment]
@MarkusShepherd reports that the Dockerfile (and `docker-compose`) works for him.

I am thinking this might have to do with my local Docker installation, as there isn't yet official Docker packaging for Ubuntu 19.10:

https://github.com/docker/for-linux/issues/832

File: api/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+CMD ["pipenv", "run", "python3", "-m", "api"]
```
Comment (MarkusShepherd) [inline api/Dockerfile]: We live in a Python 3 world here, so why not default to `python` and forget, Python 2 and that whole transition mess ever existed? 😉

File: api/docker-compose.debug.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+version: '2.1'
+
+services:
+  api:
+    image: sustainable-mobility-api
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - 8080:8080
```
Comment (MarkusShepherd) [inline api/docker-compose.debug.yml]: Do we need this file? I think in their current form the two files are equivalent anyways.

File: api/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,15 @@
+FROM python:3.7
```
Comment (huksley) [inline api/Dockerfile]: Any specific reason not to use Python 3.8?

**Comment by huksley** [comment]
> @MarkusShepherd reports that the Dockerfile (and docker-compose) works for him.

Yes, it works for me too.

**Comment by brylie** [comment]
@MarkusShepherd I made the requested changes.

File: api/Dockerfile (hunk 3 above)
Comment (brylie) [reply api/Dockerfile]: I think the HTTP API and Python estimator library can have separate release cycles, although I don't have a strong opinion here. @MarkusShepherd, thoughts?

File: api/Dockerfile (hunk 7 above)
Comment (MarkusShepherd) [reply api/Dockerfile]: I'd say at this point in time Python 3.8 is still quite new with limited support by some libraries, while it offers few new language features that encourage upgrading.

File: api/Dockerfile (hunk 3 above)
Comment (MarkusShepherd) [reply api/Dockerfile]: Yeah, I've been thinking before how to handling versioning, including tags etc. Maybe it's worth keeping them in sync eventually...

File: api/Dockerfile (hunk 7 above)
Comment (brylie) [reply api/Dockerfile]: Ah, so should I use `python:3.7-slim` or `python:3.8-slim`?

**Comment by brylie** [comment]
> Any way to fix servers entry which is generated as empty now?
> 
> image

I'm not sure how to set the default servers, nor what value to use. What do you recommend @huksley?

File: api/Dockerfile (hunk 3 above)
Comment (brylie) [reply api/Dockerfile]: OK, let's defer that decision until we have more clarity.

**Pull request merged**